### PR TITLE
feature/add_viewarea_shortcut_close_other_right

### PR DESCRIPTION
**Pull request by feloxx**
Add viewarea `CloseOtherTabs ` and `CloseTabsToTheRight ` shortcut.

In `ViewArea::setupShortcuts` impl 

- CloseOtherTabs default `""`
- CloseTabsToTheRight default `""`

### CODE REVIEW

File: src/data/core/vnotex.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
             "NewNote" : "Ctrl+Alt+N",
             "NewFolder" : "Ctrl+Alt+S",
             "CloseTab" : "Ctrl+G, X",
-            "CloseOtherTabs" : "",
-            "CloseTabsToTheRight" : "",
+            "CloseOtherTabs" : "Ctrl+G, F",
```
Comment (tamlok) [inline src/data/core/vnotex.json:19]: Do not add default values for these two shortcuts.

File: src/widgets/viewarea.cpp [hunk 2]
```diff
@@ -819,6 +820,49 @@ void ViewArea::setupShortcuts()
         }
     }
 
+    // CloseOtherTabs
+    {
+        auto shortcut = WidgetUtils::createShortcut(coreConfig.getShortcut(CoreConfig::CloseOtherTabs), this);
```
Comment (tamlok) [inline src/widgets/viewarea.cpp:825]: Can we utilize the logics of ViewSplit?

**Review by tamlok — CHANGES_REQUESTED** [review]

File: src/data/core/vnotex.json (hunk 1 above)
Comment (feloxx) [reply src/data/core/vnotex.json:19]: done

File: src/widgets/viewarea.cpp (hunk 2 above)
Comment (feloxx) [reply src/widgets/viewarea.cpp:834]: done

File: src/widgets/viewarea.cpp [hunk 3]
```diff
@@ -430,6 +431,14 @@ ViewWindow *ViewArea::getCurrentViewWindow() const
     return nullptr;
 }
 
+int ViewArea::getCurrentViewWindowIndex()
+{
+    auto split = getCurrentViewSplit();
+    if (split) {
+        return split->getCurrentViewWindowIndex();
+    }
```
Comment (tamlok) [inline src/widgets/viewarea.cpp:439]: No return value if `split` is `nullptr`?

Anyway, this function is unnecessary. ViewSplit knows its current window index, so ViewArea does not need to tell him.

File: src/widgets/viewsplit.h [hunk 4]
```diff
@@ -70,6 +72,15 @@ namespace vnotex
 
         void updateStateToWorkspace() const;
 
+        enum CloseTabMode {
```
Comment (tamlok) [inline src/widgets/viewsplit.h:75]: Use `enum class`.

File: src/widgets/viewsplit.h [hunk 5]
```diff
@@ -70,6 +72,15 @@ namespace vnotex
...
+            All,
+            Other,
```
Comment (tamlok) [inline src/widgets/viewsplit.h:77]: OtherTabs

Comment (feloxx) [reply src/widgets/viewsplit.h:77]: done

File: src/widgets/viewsplit.h [hunk 6]
```diff
@@ -70,6 +72,15 @@ namespace vnotex
...
+            Left,
```
[the remark this answers is not in the document]
Comment (feloxx) [reply src/widgets/viewsplit.h:78]: I refer to other products, they all have this, so I also want my baby to have it. plase~

File: src/widgets/viewsplit.h [hunk 7]
```diff
@@ -70,6 +72,15 @@ namespace vnotex
...
+            Right
```
[the remark this answers is not in the document]
Comment (feloxx) [reply src/widgets/viewsplit.h:79]: done

File: src/widgets/viewsplit.h (hunk 4 above)
Comment (feloxx) [reply src/widgets/viewsplit.h:75]: done

File: src/widgets/viewsplit.h [hunk 8]
```diff
@@ -70,6 +72,15 @@ namespace vnotex
...
+        };
+
+        void closeMultipleTabs(int p_idx, CloseTabMode);
```
[the remark this answers is not in the document]
Comment (feloxx) [reply src/widgets/viewsplit.h]: done

File: src/widgets/viewarea.cpp (hunk 3 above)
Comment (feloxx) [reply src/widgets/viewarea.cpp]: done

File: src/widgets/viewsplit.cpp [hunk 9]
```diff
@@ -707,45 +707,47 @@ void ViewSplit::closeTab(int p_idx)
...
              windowsNeedToClose.push_back(getViewWindow(i));
         }
+        break;
+    default:
+        break;
     }
 
-    for (auto win : windowsNeedToClose) {
-        emit viewWindowCloseRequested(win);
+    if (!windowsNeedToClose.isEmpty()) {
```
Comment (tamlok) [inline src/widgets/viewsplit.cpp]: unnecessary

Comment (feloxx) [reply src/widgets/viewsplit.cpp]: Done.

**Pull request merged**